src/core/ddsc/src/dds_writer.c

```diff
@@ -395,6 +398,12 @@ dds_entity_t dds_create_writer (dds_entity_t participant_or_publisher, dds_entit
   dds_topic_allow_set_qos (tp);
   dds_topic_unpin (tp);
   dds_publisher_unlock (pub);
+
+  // start async thread if not already started and the latency budget is non zero
+  if (async_mode && !gv->sendq_running) {
```

Comment:
There's a race condition that might cause it to start two send threads if you create two writers in parallel. Adding a mutex should do the trick.

Reply:
Fixed this